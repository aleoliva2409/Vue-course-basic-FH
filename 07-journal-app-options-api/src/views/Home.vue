<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png" />
    <button @click="goToDaybook">Go to daybook</button>
  </div>
</template>

<script>
// @ is an alias to /src

export default {
  methods: {
    goToDaybook() {
      this.$router.push({ name: 'daybook' });
    },
  },
};
</script>
