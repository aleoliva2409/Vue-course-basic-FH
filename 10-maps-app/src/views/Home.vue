<script lang="ts">
import { defineComponent } from 'vue';
import { usePlacesStore } from '../composables';
import MapView from '../components/MapView/MapView.vue';
import ButtonLocation from '../components/ButtonLocation.vue';
import SearchBar from '../components/SearchBar/SearchBar.vue';
import TripDetails from '../components/TripDetails.vue';

export default defineComponent({
  name: 'Home',
  components: { MapView, ButtonLocation, SearchBar, TripDetails },
  setup() {
    usePlacesStore();
  },
});
</script>

<template>
  <MapView />
  <ButtonLocation />
  <SearchBar />
  <TripDetails />
</template>
