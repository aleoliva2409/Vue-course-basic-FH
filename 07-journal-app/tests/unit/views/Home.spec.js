import { shallowMount } from '@vue/test-utils';
import Home from '@/views/Home';

describe('Tests on HomeView', () => {
  test('should render correctly', () => {
    const wrapper = shallowMount(Home);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should redirect to no-entry when button is clicked', () => {
    const mockRouter = {
      push: jest.fn(),
    };

    // ? se puede mockear las variables de VUE como $event, $router, $emit
    const wrapper = shallowMount(Home, {
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });

    wrapper.find('button').trigger('click');

    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' });
  });
});
