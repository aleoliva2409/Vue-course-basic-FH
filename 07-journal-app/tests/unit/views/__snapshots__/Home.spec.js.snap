// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on HomeView should render correctly 1`] = `<div class="home"><img alt="Vue logo" src=""><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button></div>`;
